lib/jsdom/living/attributes/Attr-impl.js

```diff
@@ -1,9 +1,10 @@
 "use strict";
 
-const NodeImpl = require("../nodes/Node-impl.js");
+const NodeImpl = require("../nodes/Node-impl.js").implementation;
+const { setAnExistingAttributeValue } = require("../nodes/Node-impl.js");
```

Comment:
Duplicate&nbsp;imports are&nbsp;yuck.

Reply:
Comments like this make it rather frustrating to work with you and discourage me from doing so going forward.

Reply:
Well, they&nbsp;require searching&nbsp;through Node’s&nbsp;`require`&nbsp;cache, which&nbsp;adds an&nbsp;unnecessary performance&nbsp;hit, when&nbsp;you&nbsp;can&nbsp;just&nbsp;do:

```js
const {
	implementation: NodeImpl,
	setAnExistingAttributeValue
} = require("../nodes/Node-impl.js");
```

Which&nbsp;requires&nbsp;doing so&nbsp;only&nbsp;once.

Reply:
I wish you would stop making performance claims without measurements of how much they affect real-world programs.

Reply:
_**Every**_&nbsp;`require(…)`&nbsp;call has&nbsp;some performance&nbsp;overhead, since&nbsp;**Node** has&nbsp;to&nbsp;search its&nbsp;cache, which&nbsp;has an&nbsp;<code>O(log&nbsp;<var>n</var>)</code> complexity&nbsp;at&nbsp;best, so&nbsp;it&nbsp;stands to&nbsp;reason that&nbsp;avoiding duplicate&nbsp;requires will&nbsp;keep the&nbsp;startup time&nbsp;lower&nbsp;than if&nbsp;duplicate&nbsp;requires are&nbsp;used.

Reply:
I fundamentally disagree with your approach to performance, and this further increases my doubt about working with you. I can only say that this unprofessional approach would not be accepted at major software development companies.

Reply:
1. In software engineering, it’s not _polite_ to impose your own programming style onto a project of which you are not the creator or maintainer. This is especially true in open source, which provides you with the _privilege_ (not right) of contributing.

2. Node.js uses a hash table as the cache, so a lookup is generally constant time. Even if a file system lookup is required, that would be O(_N_) time where _N_ is the level of file directory needed to traverse to find the resulting file. Here, _N_ is a constant anyway, so it’s O(1) there too.

3. We welcome performance improvements that have to do with requires, but we are also data-oriented. We want to see hard numbers before abandoning our established style. See https://github.com/jsdom/jsdom/pull/2783 for an excellent investigation and PR done by @pmdartus that provides a noticeable start-up speed-up.

4. It is an established style of ours to not use object pattern destructuring where the source property name is different from the bound identifier.

Due to these four reasons, we ask you to change this back. I also hope you may be more receptive to such review comments in the future, not just in our project but also in others. Feel free to contact me through IRC (I’m on Freenode #jsdom) or email if you have any questions.